### PR TITLE
Zmq auth

### PR DESCRIPTION
**Pull request by montjoie**
Hello

This is a PR for adding ZMQ encryption to LAVA.
Note that this is based on the postgres_pr PR

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -215,6 +215,10 @@ masters:
  - name:  lava-master	name of the master
     host: name		name of the host running lava-master (default to "local")
     webadmin_https:	Does the LAVA webadmin is accessed via https
+    zmq_auth: True/false	Does the master requires ZMQ authentication.
```
Comment (khilman) [inline README.md]: s/false/False/  ?   

I can't remember if the YAML parser is case-sensitive, but the doc should be consistent (e.g. the persistent_db entry just below uses upper case)

File: lavalab-gen.py [hunk 2]
```diff
@@ -74,18 +74,20 @@
 """)
 
 def main():
+    need_zmq_auth_gen = False
     fp = open(boards_yaml, "r")
     workers = yaml.load(fp)
     fp.close()
 
     os.mkdir("output")
+    zmq_auth_genlist = open("zmqauth/zmq_auth_gen/zmq_genlist", 'w')
```
Comment (khilman) [inline lavalab-gen.py:83]: Why is this here?  shouldn't it be later and conditional based on ZMQ usage?

**Comment by khilman** [comment]
Can you rebase on master, where the persistent db PR is already merged?

File: README.md (hunk 1 above)
Comment (montjoie) [reply README.md]: done

File: lavalab-gen.py (hunk 2 above)
Comment (montjoie) [reply lavalab-gen.py:83]: It's a list of "for which host we need to generate a key" and so it must be reseted early.
I could put it later, but I would need to detect "does it was already reseted ?" and so make it more complex.

Comment (khilman) [reply lavalab-gen.py:83]: ok, I see.

File: zmqauth/zmq_auth_fill.sh [hunk 3]
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
+
+cd $(dirname $0)
+id -u > zmq_auth_gen/id
+docker-compose build || exit $?
+docker-compose up || exit $?
```
Comment (khilman) [inline zmqauth/zmq_auth_fill.sh:6]: Can you add a `docker-compose rm` here so it doesn't leave an exited container around?

File: zmqauth/zmq_auth_gen/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,29 @@
+FROM bitnami/minideb:stretch
+
+RUN apt-get update
+
+#RUN DEBIAN_FRONTEND=noninteractive apt-get -y install postgresql
+RUN DEBIAN_FRONTEND=noninteractive apt-get -y install python3-zmq
```
Comment (khilman) [inline zmqauth/zmq_auth_gen/Dockerfile]: Please remove all the commented lines.

**Review by khilman — CHANGES_REQUESTED** [review]

**Review by khilman — APPROVED** [review]

**Pull request merged**